### PR TITLE
Add the Prow configuration for go-template-utils

### PR DESCRIPTION
**Pull request by mprahl**

### CODE REVIEW

File: core-services/prow/02_config/open-cluster-management/go-template-utils/OWNERS [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# DO NOT EDIT; this file is auto-generated using https://github.com/openshift/ci-tools.
+# See the OWNERS docs: https://git.k8s.io/community/contributors/guide/owners.md
```
Comment (dhaiducek) [inline core-services/prow/02_config/open-cluster-management/go-template-utils/OWNERS:2]: I'm not sure what the effect of leaving these lines in--it's probably safer to remove them for now and let the bot add them in.

Comment (mprahl) [reply core-services/prow/02_config/open-cluster-management/go-template-utils/OWNERS]: Sure, no problem.

**Review by dhaiducek — APPROVED** [review]

**Comment by mprahl** [comment]
@dhaiducek thanks for the review! I had to remove a new line in the Prow config for the CI to pass so your approval was removed.

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**